src/main.rs

```diff
@@ -367,14 +367,12 @@ mod cmd {
                 .read_line(&mut raw)
                 .wrap_err("Failed to read token")?;
 
-            raw = raw.trim().to_owned();
-
-            raw
+            RegistryToken::from_str(raw.trim())?
```

Comment:
I (personally) find `.parse()` more idiomatic – often you can get around the turbofish syntax if you parse a value thats an arg to something thats typed.

So here `Artifactory::new` is typed so you could get around explicitly calling from_str

Reply:
I already removed the whole type, so this is moot :)